core/src/main/java/hudson/model/AsyncPeriodicWork.java

```diff
@@ -73,8 +124,54 @@ public void run() {
     }
 
     protected StreamTaskListener createListener() {
+        File f = getLogFile();
+        if (!f.getParentFile().isDirectory()) {
+            if (!f.getParentFile().mkdirs()) {
+                logger.log(getErrorLoggingLevel(), "Could not create directory {0}", f.getParentFile());
+            }
+        }
+        if (f.isFile()) {
+            if ((lastRotateMillis + logRotateMillis < System.currentTimeMillis())
```

Comment:
This looks very similar to what is implemented in `AsyncAperiodicWork` can it be moved to some common ancestor or a utility method somewhere for better reuse than having to know there are two places to fix in the future?